specs/HostResourceMapping.md

```diff
@@ -0,0 +1,108 @@
+# Background
+We have received feedbacks that there is a need to access local files in WebView2. In this document we describe the APIs for this scenario. We'd appreciate your feedback.
+
+
+# Description
+To access local files in WebView2, define a virutal host name to local folder mapping using the `AddVirtualHostNameToFolderMapping` API,
+and then refer to local files under the folder with a normal http/https url with the virtual host name.
+
+# Examples
+Suppose the app has index.html file next to its exe file. The following code snippet demonstrates how to define a mapping
+for `app-file.invalid` host name so that it can be accessed from pages using normal http/https url like https://app-file.invalid/index.html.
+
+## Win32 C++
+```cpp
+webview2->AddVirtualHostNameToFolderMapping(
+    L"app-file.invalid", L".", COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS);
+webview2->Navigate("https://app-file.invalid/index.html");
+```
+
+## .Net and WinRT
+```c#
+webView.CoreWebView2.AddVirtualHostNameToFolderMapping(
+    "app-file.invalid", ".", CoreWebView2HostResourceAccessKind.DenyCors);
```

Comment:
Suppose my local DNS default suffix is `.contoso.com`. How does this interact with unqualified names? Suppose I do this:

```cs
if (addLocal) webView.AddVirtualHostNameToFolderMapping("litware", "A", ...);
if (addFQDN) webView.AddVirtualHostNameToFolderMapping("litware.contoso.com", "B", ...);
```


| addLocal | addFQDN | `http://litware` | `http://litware.contoso.com` |
|-|-|-|-|
| no | no | hits the wire | hits the wire |
| no | yes | ? | B |
| yes | no | A | ? |
| yes | yes | ? | ? |

Reply:
This is one reason I am not convinced that literal hostnames is a good direction for us to go in.   I'd recommend using a scheme name for this purpose instead of a hostname.  Dave R. and I are having an offline thread about this.  Either way, the rationale should be added to the PR.

Reply:
I don't think we know the answer. We will validate and document how this works.

Reply:
Updated with detailed behavior. Trailing '.' matters and not DNS resolution.